### PR TITLE
DrivAerML: 4L/384d + eta_min=1e-5 — non-zero LR floor

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Non-zero LR floor prevents complete learning rate collapse at cosine troughs, keeping gradients flowing during the trough phase. The 4L/384d run (PR #2602) showed late-epoch oscillation — epoch 141=10.2%, epoch 144=5.7% — suggesting the cosine troughs are hitting zero LR and allowing temporary loss of progress. Setting `eta_min=1e-5` means the LR never drops below 2% of its peak (5e-4 * 0.02 = 1e-5), which may smooth the trough behavior and produce more consistent convergence.

## Instructions

Run the following command exactly (one addition to base: `--cosine-eta-min 1e-5`):

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --cosine-eta-min 1e-5 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 384 \
  --model-heads 6 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --agent norman \
  --wandb-name "norman/drivaerml-4L384d-etamin" \
  --wandb-group "drivaerml-4L384d-compounds"
```

**Single addition from base:** `--cosine-eta-min 1e-5` (all other hyperparameters identical to PR #2602)

## Baseline

Current DrivAerML best (PR #2602):
- **val_primary/surface_rel_l2_pct:** 5.73% (epoch 144)
- **Baseline W&B run:** 7ogfs7ph (151 epochs, 180-min budget)
- **Architecture:** 4L/384d/6H + Fourier + no-EMA + AdamW lr=5e-4 + T_max=30
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 384 --model-heads 6 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`
- **External target:** <3.71% (AB-UPT)

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT norman:

## Results

### DrivAerML: 4L/384d + eta_min=1e-5

| Metric | Baseline (PR #2602) | This run | Delta |
|--------|-------------------|----------|-------|
| val_primary/surface_rel_l2_pct | 5.73% (epoch 144) | **11.42%** (epoch 45) | +5.69 pp |
| test_primary/surface_rel_l2_pct | N/A | **11.82%** | — |
| Training loss (final) | N/A | 0.033 | — |
| Epochs | 151 (180-min budget) | 45 (30-min timeout) | 3.4x fewer |

**⚠️ Training budget mismatch:** This run was limited to 30 min / 50 epochs (SENPAI_TIMEOUT_MINUTES=30, SENPAI_MAX_EPOCHS=50). The baseline ran for 180 min / 151 epochs. The comparison is not apples-to-apples.

### Command used
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --cosine-eta-min 1e-5 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 384 --model-heads 6 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --agent norman --wandb-name "norman/drivaerml-4L384d-etamin" \
  --wandb-group "drivaerml-4L384d-compounds"
```

### W&B run
`pgdfdstn` — https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/pgdfdstn

### What happened
- Added `cosine_eta_min` config field to train.py (default 0.0, preserving existing behavior) and passed it to `CosineAnnealingLR(eta_min=...)`.
- The eta_min=1e-5 floor means LR never drops below 2% of peak. The cosine trough behavior is clearly visible: epochs 15, 22, 29, 38, 45 consistently produce the best metrics each cycle.
- Model was **still converging strongly at cutoff** — best val dropped from 15.35% (epoch 29) → 13.12% (epoch 38) → 11.42% (epoch 45), roughly 1-2%/cycle.
- At 45 epochs, this run reached 11.42%, which beats the old 4L/256d baseline (11.97%) but cannot be compared to the 4L/384d baseline (5.73% at 151 epochs) due to the 3.4x training budget gap.
- Convergence envelope was descending at ~0.5-1%/cosine cycle, suggesting with the baseline's 180-min budget this could push well below 11%.

### Suggested follow-ups
1. **Re-run with 180-min budget** to get a fair comparison against the 5.73% baseline. The convergence rate suggests this could reach 5-7% range with full training.
2. **Try eta_min=1e-6 or 5e-6** — a smaller floor may provide marginal benefit while keeping gradients non-zero.
3. **Combine with 600 batches/epoch** (from PR #2645) — more data per epoch could accelerate the convergence rate.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2685 because ADVISOR: Closing — 11.42% at 45ep (30-min). eta_min=1e-5 cannot be evaluated without full 180-min training where the cosine schedule actually reaches the floor. norman, please pick up your next assignment..
